force-app/main/default/classes/AccountModelSettingsVMapper.cls

```diff
@@ -58,27 +58,81 @@ public virtual with sharing class AccountModelSettingsVMapper {
     }
 
     /*****************************************************************************
-    * @description Retrieves an instance of the AccountModelSettingsService class.
-    * @return An instance of AccountModelSettingsService.
-    ******************************************************************************/
+     * @description Retrieve a model representing active Account record types available
+     * for selection in Account settings and the current values set in the Hierarchy Settings
+     * fields for: Account Processor, Administrative Account Record Type and
+     * Household Addresses Acc Record Type
+     * @return An AccountModelSettingsVModel
+     *****************************************************************************/
+    public virtual AccountModelSettingsVModel getAccountModelSettingsViewModel() {
+        // retrieve current account settings from Hierarchy settings
+        AccountModelSettingsModel currentAccountSettingsModel = this.locateAccountModelSettingsService()
+            .getAccountModelSettingsModel();
+
+        Map<Id, RecordTypeModel> activeAccountRecordTypeModelsById = this.locateRecordTypeService()
+            .getModelByIdMapForSObjectType(Account.SObjectType);
+
+        // filter out inactive record types
+        for (Id accountRecordTypeModelId : activeAccountRecordTypeModelsById.keySet()) {
+            if (activeAccountRecordTypeModelsById.get(accountRecordTypeModelId).isActive == false) {
+                activeAccountRecordTypeModelsById.remove(accountRecordTypeModelId);
+            }
+        }
```

Comment:
As we're going to be performing this in multiple places, this is a good candidate to move to the RecordTypeService, which should instantiate a new map and use map.put() only on active record type models.